src/rustdoc-internals/rustdoc-json-test-suite.md

```diff
@@ -0,0 +1,47 @@
+# The `rustdoc-json` test suite
+
+This page is specifically about the test suite named `rustdoc-json`, which tests rustdoc's [json output].
+For other test suites used for testing rustdoc, see [Rustdoc tests](../rustdoc.md#tests).
+
+Tests are run with compiletest, and have access to the usuall set of [directives](../tests/directives.md).
+Frequenly used directives here are:
+
+- [`//@ aux-build`][aux-build] to have dependencies.
+- `//@ edition: 2021` (or some other edition).
+- `//@ compile-flags: --document-hidden-items` to enable [document private items].
+
+Each crate's json output is checked by 2 programs: [jsondoclint] and [jsondocck].
+
+## jsondoclint
+
+[jsondoclint] checks that all [`Id`]s exist in the `index` (or `paths`).
+This makes sure their are no dangling [`Id`]s.
+
+<!-- TODO: It does some more things too?
+Also, talk about how it works
+ -->
+
+## jsondocck
+
+<!-- TODO: shlex, jsonpath, values, variables -->
+
+### Directives
+
+- `//@ has <path>`:: Checks `<path>` exists, i.e. matches at least 1 value.
+- `//@ !has <path>`:: Checks `<path>` doesn't exist, i.e. matches 0 values.
+- `//@ has <path> <value>`: Check `<path>` exists, and 1 of the matches is equal to the given `<value>` 
+- `//@ !has <path> <value>`: Checks `<path>` exists, but none of the matches equal the given `<value>`.
+- `//@ is <path> <value>`: Check `<path>` matches exacly one value, and it's equal to the given `<value>`.
+- `//@ is <path> <value> <value>...`: Check that `<path>` matches to exactly every given `<value>`. 
+   Ordering doesn't matter here.
+- `//@ !is <path> <value>`: Check `<path>` matches exactly one value, and that value is not equal to the given `<value>`.
+- `//@ count <path> <number>` Check that `<path>` matches to `<number>` of values.
```

Comment:
can you link to an exhaustive reference of the directives? source code is fine, just somewhere to look if a new directive gets added and i can't find it in the dev guide.